// react-toolbox/lib/list/ListItem.js
'use strict';

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.RawListItem = undefined;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _font_icon = require('../font_icon');

var _font_icon2 = _interopRequireDefault(_font_icon);

var _ListItemContent = require('./ListItemContent');

var _ListItemContent2 = _interopRequireDefault(_ListItemContent);

var _ripple = require('../ripple');

var _ripple2 = _interopRequireDefault(_ripple);

var _style = require('./style');

var _style2 = _interopRequireDefault(_style);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var ListItem = function (_React$Component) {
  _inherits(ListItem, _React$Component);

  function ListItem() {
    var _Object$getPrototypeO;

    var _temp, _this, _ret;

    _classCallCheck(this, ListItem);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_Object$getPrototypeO = Object.getPrototypeOf(ListItem)).call.apply(_Object$getPrototypeO, [this].concat(args))), _this), _this.handleClick = function (event) {
      if (_this.props.onClick && !_this.props.disabled) {
        _this.props.onClick(event);
      }
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  _createClass(ListItem, [{
    key: 'renderContent',
    value: function renderContent() {
      var _ClassNames;

      var className = (0, _classnames2.default)(_style2.default.item, (_ClassNames = {}, _defineProperty(_ClassNames, _style2.default.withLegend, this.props.legend), _defineProperty(_ClassNames, _style2.default.disabled, this.props.disabled), _defineProperty(_ClassNames, _style2.default.selectable, this.props.selectable), _ClassNames), this.props.className);

      return _react2.default.createElement(
        'span',
        { className: className },
        this.props.leftIcon ? _react2.default.createElement(_font_icon2.default, { className: _style2.default.icon + ' ' + _style2.default.left, value: this.props.leftIcon }) : null,
        this.props.avatar ? _react2.default.createElement('img', { className: _style2.default.avatar, src: this.props.avatar }) : null,
        _react2.default.createElement(_ListItemContent2.default, { caption: this.props.caption, legend: this.props.legend }),
        this.props.rightIcon ? _react2.default.createElement(_font_icon2.default, { className: _style2.default.icon + ' ' + _style2.default.right, value: this.props.rightIcon }) : null
      );
    }
  }, {
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        'li',
        { className: _style2.default.listItem, onClick: this.handleClick, onMouseDown: this.props.onMouseDown },
        this.props.to ? _react2.default.createElement(
          'a',
          { href: this.props.to },
          this.renderContent()
        ) : this.renderContent(),
        this.props.children
      );
    }
  }]);

  return ListItem;
}(_react2.default.Component);

ListItem.propTypes = {
  avatar: _react2.default.PropTypes.string,
  caption: _react2.default.PropTypes.string.isRequired,
  children: _react2.default.PropTypes.any,
  className: _react2.default.PropTypes.string,
  disabled: _react2.default.PropTypes.bool,
  leftIcon: _react2.default.PropTypes.any,
  legend: _react2.default.PropTypes.string,
  onClick: _react2.default.PropTypes.func,
  rightIcon: _react2.default.PropTypes.any,
  ripple: _react2.default.PropTypes.bool,
  selectable: _react2.default.PropTypes.bool,
  to: _react2.default.PropTypes.string
};
ListItem.defaultProps = {
  disabled: false,
  ripple: false,
  selectable: false
};
exports.default = (0, _ripple2.default)({
  className: _style2.default.ripple,
  centered: false
})(ListItem);
exports.RawListItem = ListItem;
